### PR TITLE
feat(ko): Add ko builder to local artifact builder

### PR DESCRIPTION
**Pull request by halvards**
**Description**
Call the ko builder from the local artifact builder and dependency graph resolver.

**User facing changes**
None. Config file unmarshalling still skips ko config.

Tracking: #6041

### CODE REVIEW

File: pkg/skaffold/build/local/types_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import (
+	"fmt"
+	"testing"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/ko"
+	latestV1 "github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest/v1"
+	"github.com/GoogleContainerTools/skaffold/testutil"
+)
+
+func Test_newPerArtifactBuilder(t *testing.T) {
```
Comment (briandealwis) [inline pkg/skaffold/build/local/types_test.go:28]: We've generally been just capitalizing the `N` as in `TestNewPerArtifactBuilder`.

File: pkg/skaffold/graph/dependencies_test.go [hunk 2]
```diff
@@ -55,3 +58,41 @@ func TestSourceDependenciesCache(t *testing.T) {
 		}
 	})
 }
+
+func Test_sourceDependenciesForArtifact(t *testing.T) {
```
Comment (briandealwis) [inline pkg/skaffold/graph/dependencies_test.go:62]: And we'd use `TestSourceDependenciesForArtifact`

**Review by briandealwis — APPROVED** [review]

File: pkg/skaffold/build/local/types_test.go (hunk 1 above)
Comment (halvards) [reply pkg/skaffold/build/local/types_test.go]: Done

File: pkg/skaffold/graph/dependencies_test.go (hunk 2 above)
Comment (halvards) [reply pkg/skaffold/graph/dependencies_test.go]: Done

**Pull request merged**